core/trino-main/src/main/java/io/trino/server/ui/FixedUiAuthenticatorModule.java

```diff
@@ -18,14 +18,25 @@
 import com.google.inject.Scopes;
 
 import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.airlift.jaxrs.JaxrsBinder.jaxrsBinder;
 
 public class FixedUiAuthenticatorModule
         implements Module
 {
+    private final boolean isPreviewEnabled;
+
+    public FixedUiAuthenticatorModule(boolean isPreviewEnabled)
+    {
+        this.isPreviewEnabled = isPreviewEnabled;
+    }
+
     @Override
     public void configure(Binder binder)
     {
         binder.bind(WebUiAuthenticationFilter.class).to(FixedUserWebUiAuthenticationFilter.class).in(Scopes.SINGLETON);
         configBinder(binder).bindConfig(FixedUserWebUiConfig.class);
```

Comment:
I think it would be better to move these to a PreviewUiModule that binds these conditionally:

```
class PreviewUiModule 
      extends AbstractConfigurationAwareModule
{
    String authentication = buildConfigObject(WebUiAuthenticationConfig.class).getAuthentication()

    switch (authentication) {
        // bind resources according to a type
    }
}
```

Reply:
can we consider it fixed by your change with the simplified binding?